src/lib/Translation/Extractor/PolicyTranslationExtractor.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformAdminUi\Translation\Extractor;
+
+use JMS\TranslationBundle\Model\MessageCatalogue;
+use JMS\TranslationBundle\Model\Message;
+use JMS\TranslationBundle\Translation\ExtractorInterface;
+
+/**
+ * Generates translation strings for limitation types.
+ */
+class PolicyTranslationExtractor implements ExtractorInterface
+{
+    const MESSAGE_DOMAIN = 'forms';
+    const MESSAGE_ID_PREFIX = 'role.policy.';
+    const ALL_MODULES = 'all_modules';
+    const ALL_FUNCTIONS = 'all_functions';
+    const ALL_MODULES_ALL_FUNCTIONS = 'all_modules_all_functions';
+
+    /** @var array */
+    private $policyMap;
+
+    /** @var \JMS\TranslationBundle\Model\MessageCatalogue */
+    private $messageCatalogue;
+
+    /**
+     * @param array $policyMap
+     */
+    public function __construct(array $policyMap)
+    {
+        $this->policyMap = $policyMap;
+        $this->messageCatalogue = new MessageCatalogue();
+    }
+
+    /**
+     * @return \JMS\TranslationBundle\Model\MessageCatalogue
+     */
+    public function extract(): MessageCatalogue
```

Comment:
I would completely rewrite this method to the following form:

```php
public function extract(): MessageCatalogue
{        
    $catalogue = new MessageCatalogue();

    $catalogue->add($this->createMessage(self::ALL_MODULES, 'All modules'));
    $catalogue->add($this->createMessage(self::ALL_MODULES_ALL_FUNCTIONS, 'All modules / All functions'));

     foreach ($this->policyMap as $module => $functionList) {
        $catalogue->add($this->createMessage($module, $this->humanize($module)));
        $catalogue->add($this->createMessage(($module . '.' . self::ALL_FUNCTIONS, $this->humanize($module) . ' / All functions'));

        foreach ($functionList as $function => $limitationList) {
            $catalogue->add($this->createMessage(($module . '.' . $function, $this->humanize($module) . ' / ' . $this->humanize($function)));
        }
    }

    return $catalogue;
}
```

This allows us to avoid the mutable state in the service.